### PR TITLE
New GitHub actions

### PR DESCRIPTION
**Pull request by mattsecrest**

### CODE REVIEW

**Comment by mattsecrest** [comment]
@gravesti let's try this one. It's imperfect b/c it needs to build the image each time, but we can fix that over time.

**Comment by mattsecrest** [comment]
@gravesti green checks!

**Comment by mattsecrest** [comment]
Closes #233

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (gravesti) [inline .github/PULL_REQUEST_TEMPLATE.md:1]: We could consider adding check list here to remind to update docs and spelling etc to this template

File: .github/workflows/R-CMD-check.yaml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    runs-on: ${{ matrix.config.os }}
+    name: ${{ matrix.config.os }} (${{ matrix.config.r }})
+    strategy:
+      fail-fast: false
+      matrix:
+        config:
+          - {os: macOS-latest,   r: 'release'}
+          - {os: windows-latest, r: 'release'}
+          - {os: ubuntu-latest,   r: 'release'}
+    env:
```
Comment (gravesti) [inline .github/workflows/R-CMD-check.yaml:22]: Are all 3 platforms needed?

**Review by gravesti — APPROVED** [review]
@mattsecrest A few small ideas. I guess it's ok, let's see if this makes life easier, we can always revert back to a smaller subset of the IDR workflows.

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (mattsecrest) [reply .github/PULL_REQUEST_TEMPLATE.md:1]: @gravesti I agree, let's tackle that in a future PR. I have it on my list to get a developer guide together before April

File: .github/workflows/R-CMD-check.yaml (hunk 1 above)
Comment (mattsecrest) [reply .github/workflows/R-CMD-check.yaml:22]: > Are all 3 platforms needed?

Maybe? I'm always wary of Windows when it comes to things that require a C compiler.

**Pull request merged**